single_kernel_mongo/managers/mongodb_operator.py

```diff
@@ -491,9 +528,11 @@ def on_set_password_action(self, username: str, password: str | None = None) ->
         if user == MonitorUser:
             # Update and restart mongodb exporter.
             self.mongodb_exporter_config_manager.configure_and_restart()
-        # TODO: Rotate password.
-        if user in (OperatorUser, BackupUser):
-            pass
+        if user in (OperatorUser, BackupUser) and self.state.is_role(MongoDBRoles.CONFIG_SERVER):
+            self.config_server_manager.update_credentials(
+                user.password_key_name,
+                new_password,
+            )
```

Comment:
does this function allow shads to rotate passwords? I think there are some scenrios which we don't want to do that [in the previous iteration we check the conidtions](https://github.com/canonical/mongodb-operator/blob/677a6620dd9d59e820793e4a7f658770019eef84/src/charm.py#L714)

Reply:
The checks are run in the event handler.
You can have a look at `assert_pass_password_hook_checks` in the `mongodb_operator` to see that it does the same checks as it used to.